pkgs/development/python-modules/keyring/default.nix

```diff
@@ -1,27 +1,24 @@
 { stdenv, buildPythonPackage, fetchPypi
-, setuptools_scm, entrypoints, secretstorage
+, dbus-python, setuptools_scm, entrypoints, secretstorage
 , pytest, pytest-flake8 }:
 
 buildPythonPackage rec {
   pname = "keyring";
-  version = "18.0.1";
+  version = "19.1.0";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "67d6cc0132bd77922725fae9f18366bb314fd8f95ff4d323a4df41890a96a838";
+    sha256 = "13frfmws03jdyz9wxb4ylkvk80qiyb6a3h3sn7wx3ry97bn5li3a";
   };
 
   nativeBuildInputs = [ setuptools_scm ];
 
   checkInputs = [ pytest pytest-flake8 ];
 
-  propagatedBuildInputs = [ entrypoints ] ++ stdenv.lib.optional stdenv.isLinux secretstorage;
+  propagatedBuildInputs = [ dbus-python entrypoints ] ++ stdenv.lib.optional stdenv.isLinux secretstorage;
```

Comment:
The dependency relationship here is confusing
* https://github.com/jaraco/keyring/blob/7f87079734910dd26cd53bea9dd17049d4dd39dc/README.rst#linux

It uses secretstorage which uses `jeepney` but it's documented that we need `dbus-python` as well?
From reading https://github.com/jaraco/keyring/blob/8f88733d8b8a2a5f76d73b7f76e7aac72507591b/CHANGES.rst#921 it seems that maybe this is an artifact from the `python 2.7` implementation where `keyring` needed `dbus-python`. And this package can't support python 2.7 because it uses secretstorage.

Reply:
Without ```dbus-python```, keyring is not able to talk to ```kwallet```. From the README:
```
These recommended keyring backends are supported by the Python keyring lib:

macOS Keychain
Freedesktop Secret Service supports many DE including GNOME (requires secretstorage)
KDE4 & KDE5 KWallet (requires dbus)
Windows Credential Locker
```

And while it's not common, it is possible to run dbus on Mac which is why we don't restrict it to ```stdenv.isLinux```.

Reply:
thanks for clarifying 👍️